packages/shared/src/components/post/withPostById.tsx

```diff
@@ -0,0 +1,46 @@
+import React, { ComponentType, ReactElement, ReactNode } from 'react';
+import usePostById from '../../hooks/usePostById';
+import { Post } from '../../graphql/posts';
+
+type WrappedComponentType<
+  Props,
+  LayoutProps = unknown,
+> = ComponentType<Props> & {
+  getLayout?: (
+    page: ReactElement,
+    pageProps?: Props,
+    layoutProps?: LayoutProps,
+  ) => ReactNode;
+  layoutProps?: LayoutProps;
+};
+
+export const withPostById = <Props, LayoutProps = unknown>(
+  WrappedComponent: WrappedComponentType<Props, LayoutProps>,
+): WrappedComponentType<Props, LayoutProps> => {
+  const WithPostById = (props: Props & { post?: Post }): ReactElement => {
+    const { post } = props;
+
+    const { post: loadedPost, isLoading } = usePostById({
+      id: post?.id,
+      options: { initialData: { post } },
+    });
+
+    if (!post.id || isLoading) {
+      return null;
+    }
```

Comment:
Should we return `null` or was this temporary to prevent rendering any component that might use something we removed?

Reply:
@capJavert We should not continue as the render needs the post basically, it's just to prevent the initial render which is still null to not break the full UI.
It's pretty instant due to the initial data though.

Reply:
Ah ok so it goes through if `initialData` is passed? Thats ok then 👌